service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -145,16 +146,31 @@ public UUID createWorkspaceForClone(
       @Nullable TpsPolicyInputs policies,
       @Nullable List<String> applications,
       @Nullable String projectOwnerGroupId,
+      @Nullable SpendProfile spendProfile,
       UUID sourceWorkspaceId,
       CloningInstructions cloningInstructions,
       AuthenticatedUserRequest userRequest) {
+
+    // If there is a cloud platform associated with the source workspace, get it here.
+    Optional<DbCloudContext> optionalAzureCloudContext =
```

Comment:
This is what was going on in the old [CloneWorkspace](https://github.com/DataBiosphere/terra-workspace-manager/pull/1699/files#diff-461cf5a8cc6b98c856722e9336dd65621b9e0dbf776b15d3d419bcbeca17db64L62) flight to determine the cloud context to create.

Reply:
This would require further refactoring to address, but this is a change in behavior in that the old CloneWorkspace flight could clone an Azure cloud context and a GCP cloud context if a workspace had both. This will prioritize the Azure cloud context over the GCP cloud context. In practice, we only have workspaces with a single cloud context, but WSM does generally permit multiple cloud contexts.

I don't feel particularly strongly about continuing to support multiple cloud contexts in a workspace, but it might be worth looking at how difficult it would be to change this to continue to support it until we decide we officially want to drop support

Reply:
I’m not inclined to support multiple cloud contexts. We don’t use the functionality, and the new workspace creation flight didn’t support creating multiple clouds.

I could log here if there happen to be 2, but in reality we know we won’t hit that case. 

Note that I did not choose to support AWS, as it is my understanding that we will remove the AWS code.

Reply:
Fair enough, just wanted to make sure we were doing this intentionally

Reply:
I'll create a ticket for us to enforce a single cloud context per workspace (@aherbst-broad had also mentioned this).

Reply:
Created https://broadworkbench.atlassian.net/browse/WOR-1626